http/http.go

```diff
@@ -0,0 +1,88 @@
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// Response is the type that all the HTTP requests are going to return.
+type Response struct {
+	Code    int
+	Body    interface{}
+	Headers http.Header
+}
+
+//Request is the
+type Request struct {
```

Comment:
I don't get why we need to abstract Request/Response?

Reply:
The reason why I created `Request` was because I previously had a lot of parameters in the methods of this package (`Get`, `Post`, `do`, etc). I decided to create this struct to make that easier. True that we could probably get rid of this struct and use the `http` one, and then I have to change the package name 😜 .

About the `Response` one, I did it so the package takes responsibility of dealing with the serialisation and deserialisation of the response body (if you're passing an object to the method it will fill out that object, if not, it'll return raw bytes.